prog/base.rb

```diff
@@ -183,12 +183,20 @@ def hop(label)
   end
 
   def register_deadline(deadline_target, deadline_in)
-    return if strand.stack.first["deadline_target"] == deadline_target && Time.parse(strand.stack.first["deadline_at"].to_s) < Time.now + deadline_in
+    if strand.stack.first["deadline_at"].nil? ||
```

Comment:
Seems like you can clean this up with '(binding = expr)' 

Such forms are frequently verboten in C code bases, but I find them quite useful in Ruby.

Reply:
It got cleaned up a bit, but not much. I also put strand.stack.first to a variable to clean up a bit more. Though I'm not very happy with the end result. Do you see any other clean up opportunities?